### PR TITLE
Update targets and reference target setting algorithms

### PR DESCRIPTION
**Pull request by priyakasimbeg**
- Update targets for workloads 
- Update documentation for target setting algorithms
- Add HeavyBall momentum optimizer algo

### CODE REVIEW

File: reference_algorithms/target_setting_algorithms/imagenet_resnet/tuning_search_space.json [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
     "learning_rate": {
         "feasible_points": [
-            0.356449
+            4.131896390902391
         ]
     },
     "beta1": {
         "feasible_points": [
-            0.990086
+            0.9274758113254791
```
Comment (znado) [inline reference_algorithms/target_setting_algorithms/imagenet_resnet/tuning_search_space.json:9]: should warmup_steps below be 6999 or 7000?

File: reference_algorithms/target_setting_algorithms/imagenet_vit/tuning_search_space.json [hunk 2]
```diff
@@ -1,27 +1,27 @@
...
+            6999
         ]
     },
     "weight_decay": {
         "feasible_points": [
-            0.026595
+            0.08135402759553023
         ]
     },
     "grad_clip": {
```
Comment (znado) [inline reference_algorithms/target_setting_algorithms/imagenet_vit/tuning_search_space.json:27]: we set `"grad_clip": null,` internally for this workload, should we remove 1.0 below?

File: reference_algorithms/target_setting_algorithms/ogbg/tuning_search_space.json [hunk 3]
```diff
@@ -1,12 +1,12 @@
...
+            2.4917728606918423
         ]
     },
     "beta1": {
         "feasible_points": [
-            0.9536
+            0.9449369031171744
         ]
     },
     "warmup_steps": {
```
Comment (znado) [inline reference_algorithms/target_setting_algorithms/ogbg/tuning_search_space.json:12]: decay_steps_factor should be `0.861509027839639`

**Review by znado — APPROVED** [review]

File: algorithmic_efficiency/workloads/librispeech_conformer/workload.py [hunk 4]
```diff
@@ -131,4 +131,4 @@ def _build_dataset(self,
   @property
   def step_hint(self) -> int:
     """Max num steps the target setting algo was given to reach the target."""
-    return 100_000
+    return 200_000
```
[the remark this answers is not in the document]
Comment (priyakasimbeg) [reply algorithmic_efficiency/workloads/librispeech_conformer/workload.py:134]: Ah ok will revert this

File: reference_algorithms/target_setting_algorithms/imagenet_resnet/tuning_search_space.json (hunk 1 above)
Comment (priyakasimbeg) [reply reference_algorithms/target_setting_algorithms/imagenet_resnet/tuning_search_space.json:9]: There were a few other places that had this off by one difference compared to Naman's checked in parameters. I thought they were there for a reason but I couldn't make sense of it myself. If it's weird to you too, then I'll fix them all in this CL too.

File: reference_algorithms/target_setting_algorithms/ogbg/tuning_search_space.json (hunk 3 above)
Comment (priyakasimbeg) [reply reference_algorithms/target_setting_algorithms/ogbg/tuning_search_space.json:12]: Thanks for catching this

**Pull request merged**